sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -421,6 +421,15 @@ std::string GetErrorMessage(DWORD error)
 
     errorMessage += ": ";
     errorMessage += errorString.get();
+    // If the end of the error message is a CRLF, remove it.
+    if (errorMessage.back() == '\n')
+    {
+      errorMessage.erase(errorMessage.size() - 1);
+      if (errorMessage.back() == '\r')
+      {
+        errorMessage.erase(errorMessage.size() - 1);
+      }
+    }
   }
```

Comment:
Nice!

Do we still see a trailing/dangling "error code" after the period? I am not sure where that's being emitted from.

From the issue, note that extra ". 1":
> [2023-11-10T23:26:43.1733686Z T: 38480] ERROR : Request error.  Error Code: 12002: The operation timed out
. 1

I am assuming it is somewhere in this `GetErrorAndThrow()` call which has a defaulted `GetLastError()` param, but the ". 1" seems to come after this method returns the `errorMessage`:
https://github.com/Azure/azure-sdk-for-cpp/blob/dffa3edb2af2ab2673d0baad85cf4f49b7be9713/sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp#L1227-L1246

Reply:
Actually, it is not the response log message, but rather the request one.

Its this one, within `OnHttpStatusOperation()`, before this PR:
https://github.com/Azure/azure-sdk-for-cpp/blob/2337ee796e4ef8a48cb1d25a45a581a3c8b7bffb/sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp#L597-L602

Which this PR is fixing (line 641-642). So, we should be in a more readable state now :)